deployer/deployer-services/src/main/java/io/hyscale/deployer/services/builder/AppMetadataBuilder.java

```diff
@@ -25,28 +25,37 @@
 
 import io.hyscale.commons.utils.ResourceLabelUtil;
 import io.hyscale.deployer.core.model.AppMetadata;
-import io.kubernetes.client.openapi.models.V1Pod;
+import io.hyscale.deployer.core.model.ResourceKind;
+import io.hyscale.deployer.services.model.PodParent;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1StatefulSet;
 
 @Component
 public class AppMetadataBuilder {
 
     /**
-     * Gets app and service name from labels generated by hyscale
-     * For pods not deployed with hyscale AppMetadata with only namespace will be returned
+     * Gets app and service name from labels of pod parent generated by hyscale
+     * For pod controller not deployed with hyscale AppMetadata with only namespace will be returned
      * @param podList
      * @return List of {@link AppMetadata} containing details of deployed apps
      */
-    public List<AppMetadata> build(List<V1Pod> podList) {
-        if (podList == null) {
-            return null;
-        }
+    public List<AppMetadata> build(List<PodParent> podParentList) {
```

Comment:
It just iterates overall PodParentHandlers and call list all namespaces call. and then build it right

Reply:
It receives a list of PodParents and iterate over those, it's not fetching them. We do get parents from all namespace call before calling this method

Reply:
Yeah , fine with the parameter here. The call to fetch all parents should be responsible of iterating all the podparent handlers and call that getForAllNamespaces , in future even if we add job as an entity it should support.